LeadKit/LeadKit/Controllers/TableViewController.swift

```diff
@@ -0,0 +1,143 @@
+//
+//  TableViewController.swift
+//  LeadKit
+//
+//  Created by Ivan Smolin on 30/05/16.
+//  Copyright © 2016 Touch Instinct. All rights reserved.
+//
+
+import UIKit
+
+public class TableViewController: UITableViewController, CellsControllerProtocol {
+
+    private var heightCache: [NSIndexPath: CGFloat] = [:]
+
+    private var cellsObjectsCreators: [String: ViewsGenerator<UITableViewCell>] = [:]
+
+    private let cellCreationType: CellCreationType
+
+    private static let creationTypeKey = "CellCreationType"
+
+    // MARK: - Initialization
+
+    public init(style: UITableViewStyle, cellCreationType: CellCreationType = .OnTheFlight) {
+        self.cellCreationType = cellCreationType
+        super.init(style: style)
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        if let cellCreationType = CellCreationType(rawValue: aDecoder.decodeIntegerForKey(TableViewController.creationTypeKey)) {
+            self.cellCreationType = cellCreationType
+            super.init(coder: aDecoder)
+        } else {
+            return nil
+        }
+    }
+
+    public override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+
+        invalidateCache()
+    }
+
+    public override func encodeWithCoder(aCoder: NSCoder) {
+        super.encodeWithCoder(aCoder)
+        aCoder.encodeInteger(cellCreationType.rawValue, forKey: TableViewController.creationTypeKey)
+    }
+
+    /**
+     method which adds cells generator for cells with specified reuse identifier
+
+     - parameter cellsGenerator: cells generator
+     - parameter cellIdentifier: cell reuse identifier
+     */
+    public func registerCellsGenerator(cellsGenerator: ViewsGenerator<UITableViewCell>,
+                                       forCellsWithIdentifier cellIdentifier: String) {
```

Comment:
warning: Line Length Violation: Line should be 128 characters or less: currently 136 characters (line_length)